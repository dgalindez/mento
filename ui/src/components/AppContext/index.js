import { createContext, useEffect } from 'react';

import useGeolocation from '../../hooks/useGeolocation';
import useLocale from '../../hooks/useLocale';
import useWeatherApi from '../../hooks/useWeatherApi';

const AppContext = createContext({
  geolocation: {},
  locale: {},
  weather: {},
});

export const ContextComponent = ({ children }) => {
  const geolocation = useGeolocation();
  const locale = useLocale();
  const weather = useWeatherApi();

  useEffect(() => {
    if (!geolocation?.failed) {
      weather?.getLocationData(geolocation.lat, geolocation.lon);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [geolocation.failed, geolocation.lat, geolocation.lon]);

  return (
    <AppContext.Provider value={{ geolocation, locale, weather }}>
      {children}
    </AppContext.Provider>
  );
};

export default AppContext;
