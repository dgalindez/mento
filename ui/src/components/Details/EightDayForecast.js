import './index.css';

const EightDayForecast = () => {
  return (
    <div className="forecast">
      Forecast section!
    </div>
  );
};

export default EightDayForecast;
