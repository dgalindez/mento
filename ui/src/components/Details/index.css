.details {
  font-size: 20px;
}

.forecast {
  font-size: 10px;
}
