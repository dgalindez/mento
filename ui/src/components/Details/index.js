import EightDayForecast from './EightDayForecast';

import './index.css';

const Details = () => {
  return (
    <div className="details">
      Details page!
      This will render EightDayForecast and some datails for today
      <EightDayForecast />
    </div>
  );
};

export default Details;
