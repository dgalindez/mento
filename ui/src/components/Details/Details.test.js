import { render, screen } from '@testing-library/react';
// await waitFor(() => expect(contextData.locale.setZip).toHaveBeenCalledWith('zip'));
// import userEvent from '@testing-library/user-event';
// userEvent.click(screen.getByText('input.button'));
import { BrowserRouter } from 'react-router-dom';
import { I18nextProvider } from 'react-i18next';

import AppContext from '../AppContext';
import i18n from '../../utils/i18nTestHelper';

import Details from '.';

const mock = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mock,
 }));


describe('Details', () => {
  let contextData = {};
  const daily = [
    {
      max: 22.35,
      min: 14.75,
      date: 'Tue, 05 Jul 2022 06:00:00 GMT'
    },
    {
      max: 22.04,
      min: 13.8,
      date: 'Wed, 06 Jul 2022 00:00:00 GMT'
    },
    {
      max: 21.08,
      min: 12.85,
      date: 'Thu, 07 Jul 2022 00:00:00 GMT'
    },
    {
      max: 24.47,
      min: 12.91,
      date: 'Fri, 08 Jul 2022 00:00:00 GMT'
    },
    {
      max: 25.43,
      min: 13.72,
      date: 'Sat, 09 Jul 2022 00:00:00 GMT'
    },
    {
      max: 21.71,
      min: 17.79,
      date: 'Sun, 10 Jul 2022 00:00:00 GMT'
    },
  ];

  beforeEach(() => {
    contextData = {
      locale: {
        units: 'metric',
      },
      weather: {
        city: 'Seattle',
        country: 'US',
        daily,
        error: '',
        loading: false,
      },
    }
  });


  it('renders correctly', () => {
    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18n}>
          <AppContext.Provider value={contextData}>
            <Details />
          </AppContext.Provider>
        </I18nextProvider>
      </BrowserRouter>
    );

    expect(screen.getByRole('main')).toMatchSnapshot();
  })

  it('renders loading when data is being fetched', () => {
    contextData.weather.loading = true;

    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18n}>
          <AppContext.Provider value={contextData}>
            <Details />
          </AppContext.Provider>
        </I18nextProvider>
      </BrowserRouter>
    );

    expect(screen.getByRole('main')).toMatchSnapshot();
  })

  it('renders loading when an error is present', () => {
    contextData.weather.error = 'error';

    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18n}>
          <AppContext.Provider value={contextData}>
            <Details />
          </AppContext.Provider>
        </I18nextProvider>
      </BrowserRouter>
    );

    expect(screen.getByRole('main')).toMatchSnapshot();
  })

  it('navigates to root if no data is present', () => {
    contextData.weather.daily = [];

    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18n}>
          <AppContext.Provider value={contextData}>
            <Details />
          </AppContext.Provider>
        </I18nextProvider>
      </BrowserRouter>
    );

    expect(mock).toHaveBeenCalledWith('/');
  })
});
