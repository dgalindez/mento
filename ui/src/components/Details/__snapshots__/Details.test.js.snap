// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders correctly 1`] = `
<div
  class="m-4 mx-6"
  role="main"
>
  <a
    href="/"
  >
    <span
      class="icon is-large"
    >
      <i
        class="fas fa-arrow-left"
      />
    </span>
  </a>
  <span
    class="title"
  >
    Seattle
    , 
  </span>
  <span
    class="subtitle"
  >
    US
  </span>
  <table
    class="table m-4 mx-6"
  >
    <thead>
      <tr>
        <th>
          details.date
        </th>
        <th>
          details.max
           / 
          details.min
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          7/5/2022
        </td>
        <td>
          22.35
           
          ℃
           / 
          14.75
           
          ℃
        </td>
      </tr>
      <tr>
        <td>
          7/5/2022
        </td>
        <td>
          22.04
           
          ℃
           / 
          13.8
           
          ℃
        </td>
      </tr>
      <tr>
        <td>
          7/6/2022
        </td>
        <td>
          21.08
           
          ℃
           / 
          12.85
           
          ℃
        </td>
      </tr>
      <tr>
        <td>
          7/7/2022
        </td>
        <td>
          24.47
           
          ℃
           / 
          12.91
           
          ℃
        </td>
      </tr>
      <tr>
        <td>
          7/8/2022
        </td>
        <td>
          25.43
           
          ℃
           / 
          13.72
           
          ℃
        </td>
      </tr>
      <tr>
        <td>
          7/9/2022
        </td>
        <td>
          21.71
           
          ℃
           / 
          17.79
           
          ℃
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Details renders loading when an error is present 1`] = `
<div
  class="columns"
  role="main"
>
  <div
    class="control is-large is-loading"
  >
    <input
      class="hidden"
    />
  </div>
</div>
`;

exports[`Details renders loading when data is being fetched 1`] = `
<div
  class="columns"
  role="main"
>
  <div
    class="control is-large is-loading"
  >
    <input
      class="hidden"
    />
  </div>
</div>
`;
